addons/mrp_account/tests/test_valuation_layers.py

```diff
@@ -388,6 +389,45 @@ def test_validate_draft_kit(self):
         self.assertEqual(self.component.qty_available, 1)
         self.assertEqual(self.component.value_svl, 1424)
 
+    def test_average_cost_unbuild_with_byproducts(self):
+        """ Ensures that an unbuild for a manufacturing order using avg cost products won't
+            copy the value of the main product for every byproduct line, regardless of their real value
+        """
+        byproduct = self.env['product.product'].create({
+            'name': 'byproduct',
+            'type': 'product',
+        })
+        self.product1.product_tmpl_id.categ_id.property_cost_method = 'average'
+        byproduct.product_tmpl_id.categ_id.property_cost_method = 'average'
```

Comment:
Note that both products share the same category so it's not needed to set it again for `byproduct` but you can let it like that to make it explicit. Also, you don't have to go through `product_tmpl_id` to access the category, you can write `self.product1.categ_id.property_cost_method = 'average'`
If you don't want to write `blablabla.property_cost_method = 'average'` two times:
```suggestion
        self.product1.categ_id.property_cost_method = 'average'
        byproduct = self.env['product.product'].create({
            'name': 'byproduct',
            'type': 'product',
            'categ_id': self.product1.categ_id.id,
        })
```
or
```suggestion
        byproduct = self.env['product.product'].create({
            'name': 'byproduct',
            'type': 'product',
        })
        (self.product1 | byproduct).categ_id.property_cost_method = 'average'
```